exchanges/okx/okx_wrapper.go

```diff
@@ -520,22 +520,23 @@ func (ok *Okx) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (acc
 	if !ok.SupportsAsset(assetType) {
 		return info, fmt.Errorf("%w: %v", asset.ErrNotSupported, assetType)
 	}
-	balances, err := ok.GetBalance(ctx, "")
+	accountBalances, err := ok.GetNonZeroBalances(ctx, "")
 	if err != nil {
 		return info, err
 	}
-	currencyBalance := make([]account.Balance, len(balances))
-	for i := range balances {
-		free := balances[i].AvailBal
-		locked := balances[i].FrozenBalance
-		currencyBalance[i] = account.Balance{
-			Currency: currency.NewCode(balances[i].Currency),
-			Total:    balances[i].Balance,
-			Hold:     locked,
-			Free:     free,
+	currencyBalances := []account.Balance{}
+	for _, a := range accountBalances {
+		// Not clear from API docs why we'd get multiple records back in this array; Asked Okx for clarification.
+		for _, d := range a.Details {
+			currencyBalances = append(currencyBalances, account.Balance{
+				Currency: currency.NewCode(d.Currency),
+				Total:    d.EquityOfCurrency.Float64(),
+				Hold:     d.FrozenBalance.Float64(),
+				Free:     d.AvailableBalance.Float64(),
+			})
```

Comment:
```suggestion
	for i := range accountBalances {
		// Not clear from API docs why we'd get multiple records back in this array; Asked Okx for clarification.
		for j := range accountBalances[i].Details {
			currencyBalances = append(currencyBalances, account.Balance{
				Currency: currency.NewCode(accountBalances[i].Details[j].Currency),
				Total:    accountBalances[i].Details[j].EquityOfCurrency.Float64(),
				Hold:     accountBalances[i].Details[j].FrozenBalance.Float64(),
				Free:     accountBalances[i].Details[j].AvailableBalance.Float64(),
			})
```
I find via benchmarking that using a slice index rather than creating a temp local copy var is consistently more performant. Not super important

Reply:
That's true, but note that it's not clear if `accountBalances` (`i`) will ever range. It might.
So we could range `i` and `j` to pre-alloc `currencyBalances`' slice cap, and then use len() or something,
but that's going to eat away the optimization time anyway.

Overall I figured the simplicity was more important here.

Let me know if I misunderstood.

Reply:
@gloriousCode This comment really caught me off guard, because I was sure at a gopher conference long ago I'd seen the complete opposite both explained and demonstrated.

Obviously it was true when you benched it, but I'm confident it hasn't been true since around go 1.8 at least.

My [https://go.dev/play/p/4_0cDM6Lcnk](own benchmark) shows alias vars are always faster on go 1.17 through to 1.20:
```
BenchmarkLocalAlias-8           1000000000               0.0000017 ns/op
BenchmarkIndex-8                1000000000               0.0000022 ns/op
```

Will raise in slack for better discussion.